api/client.go

```diff
@@ -519,14 +530,14 @@ func (c *Client) SetAddress(addr string) error {
 	c.modifyLock.Lock()
 	defer c.modifyLock.Unlock()
 
-	parsedAddr, err := url.Parse(addr)
+	c.config.modifyLock.Lock()
```

Comment:
This bit should probably be reverted: parse the address, return err if any, if not then grab the config's lock and store the resulting address.

Reply:
There's more than just storing the address, though.  The ParseAddress function will also modify the **c.config.HTTPClient.Transport**.